replacing multi-node with multi-role (#844)
onflow/docs: docs/networks/network-architecture/index.md

In other words, every Flow node still participates in the validation of every transaction, but they do so only at one of the stages of validation.
They can therefore specialize—and greatly increase the efficiency—for their particular stage of focus.

> ### Flow node types and what they do
> 
> |                                          |   Node type    | Responsibility                                                                                                                              | What do the nodes of this type do?                                                                                                                              |

Reviewer: Not sure ... wondering if we should use `node role` here too for consistency:

```suggestion
### Flow node roles and what they do

|                                          |   Node role    | Responsibility                                                                                                                              | What nodes with this role of do?                                                                                                                              |
```

Author: I had set auto-merge on the PR and hence the changes went live...Will make another PR to address this comment.

Reviewer: sorry for the delayed commend. Thanks for the follow-up

Author: https://github.com/onflow/docs/pull/850
